packages/blitz-auth/src/server/auth-sessions.ts

```diff
@@ -289,8 +289,10 @@ const makeProxyToPublicData = <T extends SessionContextClass>(ctxClass: T): T =>
 export async function getBlitzContext(): Promise<Ctx> {
   try {
     const {headers, cookies} = require("next/headers")
-    const reqHeader = Object.fromEntries(headers())
-    const csrfToken = cookies().get(COOKIE_CSRF_TOKEN())
+    const cookieStore = await cookies()
```

Comment:
I would wrap this in `Promise.resolve` to maintain backward compatibility with Next.js 14 and earlier versions.